File: .old/Aclass_Python/A1095.py
# 运行超时
record_num, query_num = map(int, input().split())
records = []
for _ in range(record_num):
    record_info = list(input().split())
    id = record_info[0]
    hour, minute, second = record_info[1].split(":")
    time = int(hour) * 3600 + int(minute) * 60 + int(second)
    status = 1 if record_info[2] == 'in' else -1
    records.append((id, time, status))

records.sort(key=lambda x: (x[0], x[1]))

max_time, max_id, pre, clean_records, park_time = 0, [], records[0][0], [], 0
for i in range(record_num - 1):
    if records[i][0] == records[i + 1][0] and records[i][2] == 1 and records[i + 1][2] == -1:
        clean_records.append(records[i])
        clean_records.append(records[i+1])
        if records[i][0] != pre:
            if max_time < park_time:
                max_time = park_time
                max_id = [pre]
            elif max_time == park_time:
                max_id.append(pre)
            pre = records[i][0]
            park_time = 0
        park_time += (records[i + 1][1] - records[i][1])

if max_time < park_time:
    max_time = park_time
    max_id = [pre]
elif max_time == park_time:
    max_id.append(pre)

clean_records.sort(key=lambda x:x[1])

time_index, count = 0, 0
for _ in range(query_num):
    hour, minute, second = list(map(int, input().split(":")))
    query_time = hour * 3600 + minute * 60 + second
    while time_index < len(clean_records) and clean_records[time_index][1] <= query_time:
        count += clean_records[time_index][2]
        time_index += 1
    print(count)

for i in max_id:
    print(i, end=" ")
print("%02d:%02d:%02d" % (max_time // 3600, max_time % 3600 // 60, max_time % 60))
            
    

File: .old/Aclass_one_more_time/A1095.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;
struct Car_record{
    int time, status;
    string id;
};
bool cmp1(Car_record a, Car_record b) {
    if(a.id != b.id) return a.id < b.id;
    else return a.time < b.time;
}
bool cmp2(Car_record a, Car_record b) {return a.time < b.time;}
/*
void show(vector<Car_record> a){
    int len = a.size();
    for(int i = 0; i < len; i++) {
        cout << a[i].id << " " << a[i].time << " " << a[i].status << endl;
    }
}
*/
int main() {
    int record_num, query_num;
    cin >> record_num >> query_num;
    vector<Car_record> records;
    for(int i = 0; i < record_num; i++){
        Car_record temp;
        string tid, ttime, tstatus;
        cin >> tid >> ttime >> tstatus;
        temp.time = stoi(ttime.substr(0, 2)) * 3600 + stoi(ttime.substr(3, 2)) * 60 + stoi(ttime.substr(6, 2));
        temp.id = tid;
        if(tstatus == "in") temp.status = 1;
        else temp.status = -1;
        records.push_back(temp);
    }
    sort(records.begin(), records.end(), cmp1);
    // show(records);
    int max_time = 0, park_time = 0;
    string pre = records[0].id;
    vector<Car_record> clean_records;
    vector<string> max_id;
    for(int i = 0; i < record_num - 1; i++){
        if(records[i].id == records[i+1].id && records[i].status == 1 && records[i+1].status == -1) {
            clean_records.push_back(records[i]);
            clean_records.push_back(records[i+1]);
            if(records[i].id != pre){
                if(max_time < park_time) {
                    max_time = park_time;
                    max_id.clear();
                    max_id.push_back(pre);
                } else if(max_time == park_time) {
                    max_id.push_back(pre);
                }
                pre = records[i].id;
                park_time = 0;
            }
            park_time += (records[i + 1].time - records[i].time);
        }
    }
    if(max_time < park_time) {
        max_time = park_time;
        max_id.clear();
        max_id.push_back(pre);
    } else if(max_time == park_time) {
        max_id.push_back(pre);
    }

    sort(clean_records.begin(), clean_records.end(), cmp2);
    // show(clean_records);
    int time_index = 0, count = 0;
    for(int i = 0; i < query_num; i++){
        string ttime;
        cin >> ttime;
        int query_time = stoi(ttime.substr(0, 2)) * 3600 + stoi(ttime.substr(3, 2)) * 60 + stoi(ttime.substr(6, 2));
        while(time_index < clean_records.size() && clean_records[time_index].time <= query_time){
            count += clean_records[time_index].status;
            time_index++;
        }
        cout << count << endl;
    }
    for(int i = 0; i < max_id.size(); i++) {
        cout << max_id[i] << " ";
    }
    printf("%02d:%02d:%02d", max_time / 3600, max_time % 3600 / 60, max_time % 60);
    return 0;
}